### PR TITLE
tests: constrain proptest strategies to reduce flakiness

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR constrains some of the `ScalarType` and `Datum` proptest strategies to reduce edge case strategies that could blow up to very large sizes and cause timeouts

### Motivation

Fixes https://github.com/MaterializeInc/materialize/issues/27370
Fixes https://github.com/MaterializeInc/materialize/issues/27430

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

File: src/repr/src/scalar.rs [hunk 1]
```diff
@@ -3623,52 +3623,44 @@ impl Arbitrary for ScalarType {
...
+                    let fields_strat =
+                        prop::collection::vec((any::<ColumnName>(), column_type_strat), 0..10);
+
+                    // Now we combine it with the default strategies to get Records.
+                    (fields_strat, any::<Option<GlobalId>>())
+                        .prop_map(|(fields, custom_id)| ScalarType::Record { fields, custom_id })
+                        .boxed()
+                },
+            ])
+        })
```
Comment (ParkMyCar) [inline src/repr/src/scalar.rs:3663]: nothing changed in this entire block other than changing a `4` to `3` and removing a comment. Removing the (incorrect) comment caused the block to get reformatted

**Comment by nrainer-materialize** [comment]
Awesome, thank you!

I triggered a [Miri test.](https://buildkite.com/materialize/nightly/builds/7969)

**Review by def- — APPROVED** [review]

**Review by bkirwi — APPROVED** [review]
🙏

**Comment by ParkMyCar** [comment]
Looks like miri is still timing out, I just ignored miri for these tests since they don't use any unsafe so there isn't any value in running it anyways

**Pull request merged**